File: Construct a Binary tree from string with bracket representation/code.java
class Solution {
    static int index;
    public static Node treeFromString(String s) {
        index = 0;
        return solve(s);
    }
    public static Node solve(String str){
        if(index>=str.length()){
            return null;
        }
        StringBuilder sb = new StringBuilder("");
        while(index<str.length() && str.charAt(index)>='0' && str.charAt(index)<='9'){
            sb.append(str.charAt(index++));
        }
        Node ans = new Node(Integer.parseInt(sb.toString()));
        
        if(index<str.length() && str.charAt(index)=='('){
            index++;
            ans.left = solve(str);
        }
        if(index<str.length() && str.charAt(index)=='('){
            index++;
            ans.right = solve(str);
        }
        index++;
        return ans;
    }
}

File: Find kth smallest element in a bst/code.cpp
class Solution {

  public:

    // Return the Kth smallest element in the given BST

    int ans = -1;

    void solve(Node* root, int &K){

        if(root == NULL) return;

        solve(root->left,K);

        K--;

        if(K == 0){

            ans = root->data;

            return;

        }

        solve(root->right,K);

        

    }

    int KthSmallestElement(Node *root, int K) {

        // add code here.

        ans = -1;

        solve(root,K);

        return ans;

    }

};
